polyply/tests/test_annotate_lingads.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Test that ligands are properly annotated.
+"""
+import pytest
+import numpy as np
+import networkx as nx
+import vermouth
+import polyply
+from polyply.src.meta_molecule import Monomer, MetaMolecule
+from polyply.src.topology import Topology
+
+# <mol_name>#<mol_idx>-<resname>#<resid>
+@pytest.mark.parametrize('spec, expected', [
+    ('', {}),
+    ('A-ALA#1', {'molname': 'A', 'resname': 'ALA', 'resid': 1}),
+    ('A-ALA1', {'molname': 'A', 'resname': 'ALA1'}),
+    ('ALA1', {'molname': 'ALA1'}),
+    ('A-ALA', {'molname': 'A', 'resname': 'ALA'}),
+    ('ALA', {'molname': 'ALA'}),
+    ('ALA#5', {'molname': 'ALA', 'mol_idx':5}),
+    ('A#1-ALA#1', {'molname': 'A', 'mol_idx':1, 'resname': 'ALA', 'resid': 1.0}),
+    ('#1-ALA#1', {'mol_idx':1, 'resname': 'ALA', 'resid': 1.0}),
+    ('#2', {'mol_idx': 2}),
+    ('#2-ALA', {'mol_idx': 2, 'resname': 'ALA'}),
+    ('#2-#1', {'mol_idx': 2, 'resid':1.0}),
```

Comment:
Missing some sadistic edge cases, such as '-' in mol names, etc.
How about `#-#`? `#-#2` `A#-#` `-ALA`, etc, etc